react-app/ccf/src/Sidebar.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useState } from "react";
+import "./Sidebar.css";
+
+const Sidebar: React.FC = () => {
+  // State to handle the collapsed/expanded state
+  const [isCollapsed, setIsCollapsed] = useState(false);
+
+  // Function to toggle collapse
+  const toggleSidebar = () => {
+    setIsCollapsed(!isCollapsed);
+  };
+
+  return (
+    <div className={isCollapsed ? "sidebar collapsed" : "sidebar"}>
+      <div className="sidebar-header">
+        <button onClick={toggleSidebar} className="toggle-btn">
+          {isCollapsed ? ">>>" : "<<<"}
+        </button>
+      </div>
+      {/* Only show the menu when the sidebar is not collapsed */}
+      {!isCollapsed && (
+        <ul className="sidebar-menu">

```

Comment:
We need to be able to pass a list links to the sidebar component. This is to make it reusable across the different pages.